app/models/bus_stop.rb

```diff
@@ -1,155 +1,32 @@
 # frozen_string_literal: true
 
 require 'csv'
+include DateAndTimeMethods
 
 class BusStop < ApplicationRecord
-  include DateAndTimeMethods
   has_paper_trail
-  validates :name, presence: true
-  validates :hastus_id, presence: true, uniqueness: true
+
   has_and_belongs_to_many :routes
   has_many :bus_stop_fields
   has_many :fields, through: :bus_stop_fields
 
+  accepts_nested_attributes_for :bus_stop_fields
```

Comment:
This is the only real change in this section. Everything else I just moved around for clarity.